### ISSUE TITLE
[BUG] Path.urlFromString is using the incorrect API for iOS

### ISSUE THREAD

**Issue by wongk** [suparngp/kotlin-multiplatform-projects#63]
**Describe the bug**
Path.urlFromString is using the following API: https://developer.apple.com/documentation/foundation/nsurl/1572047-urlwithstring?language=objc. This API always returns null for plain file paths, and the documentation states: "To create NSURL objects for file system paths, use fileURLWithPath:isDirectory: instead." This means that all FileSystem APIs that are using Path.urlFromString return null for iOS targets.

**Version**
1.0.8

**Platforms affected:**
- iOS [All]

**Comment by suparngp** [comment]
Thanks for reporting. I am looking at it now.

**Comment by suparngp** [comment]
I am unable to reproduce the issue at my end. For example,
```
val path = FileSystem.contentsDirectory.absolutePath?.byAppending("hello.txt")
    if (path != null) {
        println(path.component!!)
    }
```
prints `/Users/.../Library/Developer/CoreSimulator/Devices/33248B90-E747-48B8-8FEE-DF39CD5287D5/data/Containers/Data/Application/E6D47516-A95D-4D16-8C2D-30C06AA7F4B9/Documents/hello.txt`

Internally, `byAppending()` uses `url` computed property which calls `NSURL.URLWithString`. I don't see how this may return null as `file://` is a valid URL format. As far as using `fileURLWithPath` is concerned, I deliberately choose not to use it because if a URL string contains `file://` like in the case of known directory URLs (documents, cache etc), it adds an extra `file://` and normalizes the given path to `file:/`. For example, `URL.init(fileURLWithPath: "file:///Hello")` returns `file:///file:/Hello` where as using URLWithString returns `file://Hello` as absolute URL. 

Can you please share an example of how you are using the API? I may have missed something.

**Comment by wongk** [comment]
Some of the APIs take file paths as string, i.e., `FileSystem.readDir` and `FileSystem.readFile`. These operations fail as described above.

**Comment by suparngp** [comment]
I fixed this in v1.0.9 which is now building. Please verify and close the issue if you think it is resolved.

**Comment by wongk** [comment]
Thank you, will do.
